file module should check invalid arguments (fixes #2135)

mmoya opened the pull request:
Trivial fix for #2135.

mmoya committed:
file module should check invalid arguments (fixes #2135)

mpdehaan commented:
Just quick question, did you check this with being able to set owner= on something like template or the copy module?

Want to make sure those are still passed down ok.

(Will obv test myself in a bit :))

mmoya commented:
Both `copy` and `template` are _not_ working as expected. With this applied template accepts `owner=` as well as `nonexistent=someval`. `file` bails out properly.

I'll cancel this PR as it doesn't solve the bug.

dhozac committed:
slurp: import base64 as module_common no longer does

dhozac committed:
host.get_variables includes too many variables, use host.vars instead

Not quite as attractive, but allows variable precedence and lazy
resolution to work as intended, rather than resolving them on a
per-inventory basis.

dhozac committed:
Merge pull request #2271 from MWGriffin/patch-1

Update get_url

dhozac committed:
Fix parse_kv invocation in group_by to use imported name

Fixes #2277.

mpdehaan committed:
Revert "for #2196 now gruopby module invalidates host cached variables so they can be reevaluated with new groups"

This reverts commit 2d86671aa5b0fcdabe35f88fc5e09197acd3f17d.

dhozac committed:
group_by needs the untemplated module_args

mpdehaan committed:
Revert "Revert "for #2196 now gruopby module invalidates host cached variables so they can be reevaluated with new groups""

This reverts commit dc0515a88a8aa78630f0b7118563c7a23c898839.

We didn't need to revert this :)

mpdehaan committed:
Update doc reference to clarify that $item isn't expanded here.

dhozac committed:
YAML requires quoting the entire value when it starts with a quote

Fixes #2280.

dhozac committed:
When a host list is used, no parser is present

mmoya commented:
Reopened this after rebasing `bug-fixes` and GH is now including a lot of commits, so (re)closing it. Will file another PR.
